test/e2e/upgrade_test.go

```diff
@@ -0,0 +1,81 @@
+// +build upgrade
+
+package e2e
+
+import (
+	"context"
+	"os"
+	"regexp"
+	"strings"
+	"testing"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	v1 "github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+)
+
+const (
+	envUpgradeVersionKey = "UPGRADE_TEST_VERSION"
+	upgradeTestTag       = "next"
+)
+
+type OperatorUpgradeTestSuite struct {
+	suite.Suite
+}
+
+func TestOperatorUpgrade(t *testing.T) {
+	suite.Run(t, new(OperatorUpgradeTestSuite))
+}
+
+func (suite *OperatorUpgradeTestSuite) SetupTest() {
+	t = suite.T()
+}
+
+func (suite *OperatorUpgradeTestSuite) TestUpgrade() {
+	upgradeTestVersion := os.Getenv(envUpgradeVersionKey)
+	versionRegexp := regexp.MustCompile(`^[\d]+\.[\d]+\.[\d]+`)
+
+	require.Regexp(t, versionRegexp, upgradeTestVersion,
+		"Invalid upgrade version, need to specify a version to upgrade in format X.Y.Z")
+
+	ctx, err := prepare(t)
```

Comment:
@rubenvp8510 A couple of more small things, which are more for consistently with other tests than correctness.

1. Can you move everything from the ""ctx, err := prepare()" down to "fw = framework.Global" into a SetupSuite() method, making sure to remove the "defer ctx.Cleanup() call.
2. Then add a TearDownSuite() which should look like this one: https://github.com/jaegertracing/jaeger-operator/blob/master/test/e2e/elasticsearch_test.go#L56-L58    handleSuiteTearDown will take care of the ctx.Cleanup()